### PR TITLE
add semantic-metrics implementation

### PR DESCRIPTION
**Pull request by mattnworb**
This is a port of internal code we developed at Spotify that implements Folsom's `Metrics` interface with [semantic-metrics][].

The module was written before semantic-metrics was open sourced, so it was not added to folsom at the time of writing but kept internal.

There isn't any real reason to keep it internal any longer so this change adds it to folsom so that things like versions or API changes can be managed by having this module in the same source tree as folsom itself.

To add this to this repo but keep its usage as optional for any folsom users, I restructured the project as a multimodule project first. I also removed the java8 profile from the folsom module (and moved the checkstyle plugins into the parent pom) as it seems unnecessary to have those plugins be optional now that the source is java8-only.

[semantic-metrics]: https://github.com/spotify/semantic-metrics

### CODE REVIEW

**Comment by mattnworb** [comment]
I am never sure if the `nexus-staging-maven-plugin` plugin belongs in `<plugins>` or `<pluginManagement>` for a multimodule build, I left it in the parent's `<plugins>` but I'm not sure how to test it without actually doing a release.

**Comment by mattnworb** [comment]
Ah, I see the `java8` profile was there to prevent the checkstyle plugin from being run when built with `oraclejdk9`. I'll add that back.

File: folsom-semantic-metrics/src/main/java/com/spotify/folsom/client/SemanticFolsomMetrics.java [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+import com.spotify.folsom.Metrics;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricRegistry;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * {@link com.spotify.folsom.Metrics} implementation using semantic-metrics.
+ * Mostly a port of the YammerMetrics class.
```
Comment (protocol7) [inline folsom-semantic-metrics/src/main/java/com/spotify/folsom/client/SemanticFolsomMetrics.java:33]: Aside, we should really rename that class, Yammer hasn't been involved for some time now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+### 1.1.0
+Add `folsom-semantic-metrics` module containing a [semantic-metrics][]-based implementation of folsom's `Metrics`.
```
Comment (protocol7) [inline CHANGELOG.md]: Nit: s/folsom's/Folsom's/

**Review by protocol7 — APPROVED** [review]
+1

**Comment by mattnworb** [comment]
Looks like the build status is stuck in pending because the refactor to be a multimodule build broke coveralls somehow. I'll look into this.

File: folsom-semantic-metrics/src/main/java/com/spotify/folsom/client/SemanticFolsomMetrics.java (hunk 1 above)
Comment (mattnworb) [reply folsom-semantic-metrics/src/main/java/com/spotify/folsom/client/SemanticFolsomMetrics.java:33]: how about upgrading to the latest version of dropwizard-metrics and renaming the class to that? I can do that in another PR. 

Do you think its worth moving that class into its own module too?

File: CHANGELOG.md (hunk 2 above)
Comment (mattnworb) [reply CHANGELOG.md]: fixed

**Comment by mattnworb** [comment]
Closing as I think coveralls isn't updating the build status properly if the PR is from a fork.

**Pull request closed without merging**